app/models/event.rb

```diff
@@ -242,26 +250,30 @@ def calculate_progress
   end
 
   ##
-  # Returns end of the event
+  # Returns the room in which the event is scheduled
   #
-  def end_time
-    self.start_time + self.event_type.length.minutes
+  def scheduled_room
+    event_schedule.room
   end
 
   ##
-  # Returns events that are scheduled in the same room and start_time as event
+  # Returns the start time at which this event is scheduled
   #
-  def intersecting_events
-    room.events.where(start_time: start_time).where.not(id: id)
+  def scheduled_start_time
+    event_schedule.start_time
+  end
+
+  def event_schedule(schedule_id=nil)
+    event_schedules.find_by(schedule_id: (schedule_id.present? ? schedule_id : program.try(:selected_schedule)))
```

Comment:
I would split that in 

```selected_event_schedule``` and ```event_schedule```

and as yesterday discussed, I would try to do it with an association!